pyresample/area_config.py

```diff
@@ -397,8 +397,9 @@ def create_area_def(area_id, projection, width=None, height=None, area_extent=No
     ----------
     area_id : str
         ID of area
-    projection : dict or str
-        Projection parameters as a proj4_dict or proj4_string
+    projection : pyproj CRS object, dict or str
```

Comment:
Looking at the code, in ends up calling pyproj's `CRS.from_user_input` with accept multiple other things that what is mentionned on the current line.
https://pyproj4.github.io/pyproj/dev/api/crs/crs.html#pyproj.crs.CRS.from_user_input

I was also mislead by an incorrect docstring in pyresample:
https://github.com/pytroll/pyresample/blob/master/pyresample/area_config.py#L543-L554

If you feel like fixing it also in this PR...

Reply:
Done